### PR TITLE
geojson unification of event handling

### PR DESCRIPTION
**Pull request by nmccready**
This PR gets the geojson directive on the same event argument structure for callbacks.

### CODE REVIEW

File: src/services/leafletEvents.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
 angular.module("leaflet-directive").factory('leafletEvents',
     function (leafletMapEvents, leafletMarkerEvents, leafletPathEvents) {
-        var retObj = {};
-        angular.extend(retObj,leafletMapEvents, leafletMarkerEvents, leafletPathEvents);
-        return retObj;
+      //NOTE THIS SHOULD BE DEPRECATED infavor of getting a specific events helper
```
Comment (nmccready) [inline src/services/leafletEvents.js]: This file really should be deleted as it is a connivence that causes more havoc on the development side.

File: src/directives/geojson.js [hunk 2]
```diff
@@ -36,27 +35,7 @@ angular.module("leaflet-directive")
                                 layer.bindLabel(feature.properties.description);
                             }
 
-                            layer.on({
```
Comment (nmccready) [inline src/directives/geojson.js]: Unique event handling to geojson removed.

**Comment by nmccready** [comment]
@zacronos @tombatossals

File: dist/angular-leaflet-directive.js [hunk 3]
```diff
@@ -3013,27 +3016,7 @@ angular.module("leaflet-directive")
...
-                                        $rootScope.$broadcast('leafletDirectiveMap.geojsonMouseout', e);
-                                    });
-                                },
-                                click: function(e) {
-                                    safeApply(leafletScope, function() {
-                                        $rootScope.$broadcast('leafletDirectiveMap.geojsonClick', feature, e);
-                                    });
-                                }
-                            });
+                            leafletGeoJsonEvents.bindEvents(layer, null, feature, leafletScope, null);
```
Comment (nmccready) [inline dist/angular-leaflet-directive.js]: I probably can get the layer or the nested featureName here. I need to look into this more.

**Comment by nmccready** [comment]
@jessertaylor

**Comment by zacronos** [comment]
LGTM

**Comment by nmccready** [comment]
Don't merge yet, there are issues in the example.

**Comment by nmccready** [comment]
@tombatossals @zacronos ready

**Comment by jessertaylor** [comment]
:thumbsup:

**Comment by nmccready** [comment]
@jessertaylor ping me on twitter or my gmail on github

**Pull request merged**